### PR TITLE
we now do the vagrant right?

### PR DESCRIPTION
**Pull request by zeeshanlakhani**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  # All Vagrant configuration is done here. The most common configuration
+  # options are documented and commented below. For a complete reference,
+  # please see the online documentation at https://docs.vagrantup.com.
+  config.vm.box = "ubuntu/xenial64"
+  config.disksize.size = "50GB"
+
+  if ENV['netbricks'] then
+    config.vm.synced_folder ENV["netbricks"], "/vagrant", disabled: false
```
Comment (zeeshanlakhani) [inline Vagrantfile]: @seancribbs should we do the else here so that we can default to `"."`

Comment (seancribbs) [reply Vagrantfile]: In that case I'd probably change it to:

```ruby
config.vm.synced_folder ENV["netbricks"] || '.', "/vagrant", disabled: false
```

File: Vagrantfile [hunk 2]
```diff
@@ -9,10 +9,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # please see the online documentation at https://docs.vagrantup.com.
   config.vm.box = "ubuntu/xenial64"
   config.disksize.size = "50GB"
+  config.vm.synced_folder ENV.fetch("netbricks", '.'), "/vagrant", disabled: false
```
Comment (zeeshanlakhani) [inline Vagrantfile]: dope.

File: Vagrantfile [hunk 3]
```diff
@@ -3,50 +3,31 @@
...
-  #                  -v /var/run:/var/run).join(" "),
-  #         restart: "no",
-  #         daemonize: true,
-  #         auto_assign_name: false,
-  #         cmd: "/bin/bash -c 'while true; do sleep 1; done'"
-  # end
+  config.vm.provision "shell", path: "vm-setup.sh", args: $dimage
 
-  # VirtualBox-specific configuration
+  # VirtualBox-specific configurationx
```
Comment (seancribbs) [inline Vagrantfile]: typo

File: docker.mk [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+CONTAINER ?= $(SANDBOX_CONTAINER)
+BASE_DIR ?= $(SANDBOX_BASE_DIR)
+
+ifeq ($(CONTAINER),$(DPDK_CONTAINER))
+  DOCKERFILE=dpdk/Dockerfile
+endif
+
+LINUX_HEADERS = -v /lib/modules:/lib/modules
+MOUNTS = $(LINUX_HEADERS) \
```
Comment (seancribbs) [inline docker.mk]: 👍

File: dpdk/Dockerfile [hunk 5]
```diff
@@ -8,25 +8,25 @@ ENV RTE_TARGET build
...
+    python3 \
+    wget && \
+    rm -rf /var/lib/apt /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 RUN mkdir -p /opt/dpdk
 
 WORKDIR /opt/dpdk
 
-RUN wget https://fast.dpdk.org/rel/dpdk-$VERSION.tar.xz && \
+RUN wget --no-check-certificate https://fast.dpdk.org/rel/dpdk-$VERSION.tar.xz && \
```
Comment (seancribbs) [inline dpdk/Dockerfile]: 👍

**Pull request merged**